### DISCUSSION TITLE
Is there an example application that showcases using Spring Modulith?

### DISCUSSION THREAD

**Discussion by wimdeblauwe** [spring-projects/spring-modulith#131] [Q&A] [answered]
I would like to investigate Spring Modulith in more detail. I wonder if there is an example application that showcases using the framework? I am especially interested to understand how domain objects are exposed from one module to the other. What do you expose, what do you hide?

**Comment by gkislin** [comment]
I've fond simple one: https://github.com/ajavageek/spring-modulith-sample

**Comment by odrotbohm** [comment, marked as answer]
Here's a list of publicly accessible ones I am aware of (to be continuously updated):

* [RESTBucks](https://github.com/odrotbohm/spring-restbucks) – A sample project to provide a hypermedia-based API to order coffee. Based on the book [Rest in Practice](https://www.oreilly.com/library/view/rest-in-practice/9781449383312/).
* [Salespoint](https://github.com/st-tu-dresden/salespoint) – A POS library used in the software engineering lab of the TU Dresden.
* [Spring Modulith Sample](https://github.com/ajavageek/spring-modulith-sample) – by @ajavageek, found by @gkislin.
* [Our own example](https://github.com/spring-projects/spring-modulith/tree/main/spring-modulith-examples)

**Reply by duskote** [reply]
hi @odrotbohm, the link `Our own example` is broken. I believe it should point here:
https://github.com/spring-projects/spring-modulith/tree/main/spring-modulith-examples

**Reply by odrotbohm** [reply]
Thanks for the ping. That's been fixed!

**Reply by MichaelMeijer** [reply]
Anyone had a look into distributing event listening/handling across multiple instances, e.g. running replicas of a spring modulith app ? It seems not natively supported. The instance firing the event seems to also be the one handling it.  Only when moving to eventing beyond spring modulith this seems to function properly.
